visualvm/libs.profiler/lib.profiler/src/org/graalvm/visualvm/lib/jfluid/heap/Progress.java

```diff
@@ -51,16 +51,20 @@ synchronized static void register(Listener onChange) {
         }
     }
 
-    synchronized static void notifyUpdates(Handle h, int type) {
+    private synchronized static void notifyUpdates(Handle h, Type type) {
```

Comment:
how bout ?

    private synchronized static void notifyUpdates(Handle h, BiConsumer<Listener, Handle> c) {
        for (Listener onChange : listeners) {
             c.accept(onChange, h);
         }
    }

and then call it like 

        @Override
        public void close() {
            notifyUpdates(this, 2);
            notifyUpdates(this, Listener::finished);
        }